src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -173,9 +173,6 @@ def _ensure_ffi_initialized(cls):
                     cls._legacy_provider = cls.lib.OSSL_PROVIDER_load(
                         cls.ffi.NULL, b"legacy"
                     )
-                    _openssl_assert(
-                        cls.lib, cls._legacy_provider != cls.ffi.NULL
-                    )
```

Comment:
Removing this means that if it fails to load _unintentionally_, the user will never notice. Can this happen? Should we require users to opt into "legacy provider is not required" somehow?